### PR TITLE
[Test modernization] on TopBar components tests

### PR DESCRIPTION
**Pull request by kpinzas-sh**
**WHY are these changes introduced?**

I am updating existing tests for Message, Menu, Search, SearchDismissOverlay, SearchField, UserMenu and TopBar components to use the new modern framework.

**WHAT is this pull request doing?**

Part of test modernization (https://docs.google.com/spreadsheets/d/1GBuEZbOpVYJLNISK7gL69DU8rCxocn5L5GYaKtPXPbU/edit#gid=1498187033), updating tests using {mountWithAppProvider} from 'test-utilities/legacy' to {mountWithApp} from 'test-utilities'.

### CODE REVIEW

File: src/components/TopBar/components/UserMenu/tests/UserMenu.test.tsx [hunk 1]
```diff
@@ -17,13 +16,37 @@ describe('<UserMenu />', () => {
   };
 
   it('renders with the given props', () => {
-    const userMenu = mountWithAppProvider(<UserMenu {...userMenuProps} />);
-    expect(userMenu.find(UserMenu).props()).toStrictEqual(userMenuProps);
+    const userMenu = mountWithApp(<UserMenu {...userMenuProps} />);
```
Comment (kpinzas-sh) [inline src/components/TopBar/components/UserMenu/tests/UserMenu.test.tsx:19]: This is the only big change that I am doing, but not sure if this makes sense to test that the props are being render.

File: src/components/TopBar/components/Menu/tests/Menu.test.tsx [hunk 2]
```diff
@@ -74,61 +74,66 @@ describe('<Menu />', () => {
         status: 'new' as 'new',
       },
     };
-    const menu = mountWithAppProvider(
+    const menu = mountWithApp(
       <Menu {...defaultProps} message={message} open />,
     );
 
-    expect(menu.find(Message).prop('badge')).toStrictEqual(message.badge);
+    expect(menu.find(Message)!.prop('badge')).toStrictEqual(message.badge);
```
Comment (rdott) [inline src/components/TopBar/components/Menu/tests/Menu.test.tsx:81]:
```suggestion
    expect(menu).toContainReactComponent(Message, {badge: message.badge});
```

File: src/components/TopBar/components/Menu/tests/Menu.test.tsx [hunk 3]
```diff
@@ -74,61 +74,66 @@ describe('<Menu />', () => {
...
   });
 
   describe('isFullHeight', () => {
     it('passes isFullHeight to popover as false if menu is not provided a message', () => {
       const {message, ...rest} = defaultProps;
-      const menu = mountWithAppProvider(<Menu {...rest} open />);
+      const menu = mountWithApp(<Menu {...rest} open />);
 
-      expect(menu.find(Popover).prop('fullHeight')).toBe(false);
+      expect(menu.find(Popover)!.prop('fullHeight')).toBe(false);
```
Comment (rdott) [inline src/components/TopBar/components/Menu/tests/Menu.test.tsx:128]:
```suggestion
      expect(menu).toContainReactComponent(Popover, {fullHeight: false});
```

File: src/components/TopBar/components/UserMenu/tests/UserMenu.test.tsx (hunk 1 above)
Comment (rdott) [reply src/components/TopBar/components/UserMenu/tests/UserMenu.test.tsx:19]: This makes sense to me

**Review by rdott — APPROVED** [review]
Couple of small comments, but otherwise it looks good!

**Review by BPScott — APPROVED** [review]

**Pull request merged**